sencha-workspace/SlateDemonstrationsStudent/app/controller/Dashboard.js

```diff
@@ -222,17 +219,26 @@ Ext.define('SlateDemonstrationsStudent.controller.Dashboard', {
             }
         }
 
+        // get an array of competency IDs from the studentCompetenciesStore
+        let studentCompetencyIDs = studentCompetenciesStore.getData().items.map(rec => rec.get('CompetencyID'));
 
         // build cards
         competenciesCount = competenciesStore.getCount();
         competencyIndex = 0;
 
         for (; competencyIndex < competenciesCount; competencyIndex++) {
-            cardConfigs.push({
-                competency: competenciesStore.getAt(competencyIndex)
-            });
+            let competency = competenciesStore.getAt(competencyIndex);
+
+            if (studentCompetencyIDs.includes(competency.get('ID'))) {
```

Comment:
**suggestion (non-blocking)**: Prefer the dedicated identifier getter for getting ids out of models:

```suggestion
            if (studentCompetencyIDs.includes(competency.getId())) {
```